risc0/zkvm/src/lib.rs

```diff
@@ -71,6 +71,7 @@
 
 extern crate alloc;
 
+#[cfg(target_os = "zkvm")]
```

Comment:
I think this breaks the ability to use rust-analyzer

Reply:
Thinking about this some more, the original bug report and issues we've seen internally point to the undefined `_end` symbol used in `sys_alloc_aligned`. I'm thinking that it's probably better to revert all of these changes and put cfg's around where _end appears... Let me know if you have thoughts on this. I'm going to make this change and change the title/description of this PR.